File: CSharp/PaymentsService/Infrastructure/Payments/GooglePay/ProcessGooglePayPaymentRequest.cs
using PaymentsService.Domain.Payments;

namespace PaymentsService.Infrastructure.Payments.GooglePay;

public sealed class ProcessGooglePayPaymentRequest : ProcessPaymentBaseRequest
{
    public ProcessGooglePayPaymentRequest() 
        : base("GooglePay")
    {
    }

    public string Email { get; set; }
    public string Token { get; set; }
}

File: Python/PaymentsService/Infrastructure/Payments/GooglePay/ProcessGooglePayPaymentRequest.py
from Domain.Payments.ProcessPaymentBaseRequest import ProcessPaymentBaseRequest


class ProcessGooglePayPaymentRequest(ProcessPaymentBaseRequest):
    def __init__(self, email = None, token = None, currency = None, amount = 0):
        super().__init__("GooglePay", currency, amount)
        self.email = email
        self.token = token

    def to_dict(self):
        return {
            "paymentMethod": self.paymentMethod,
            "currency": self.currency,
            "amount": self.amount,
            "token": self.token,
            "email": self.email
        }